node/pkg/p2p/ccq_p2p.go

```diff
@@ -73,67 +65,23 @@ func newCcqRunP2p(
 
 func (ccq *ccqP2p) run(
 	ctx context.Context,
-	priv crypto.PrivKey,
 	gk *ecdsa.PrivateKey,
+	h host.Host,
 	networkID string,
-	bootstrapPeers string,
-	port uint,
 	signedQueryReqC chan<- *gossipv1.SignedQueryRequest,
 	queryResponseReadC <-chan *query.QueryResponsePublication,
 	errC chan error,
 ) error {
 	var err error
-
-	components := DefaultComponents()
-	if components == nil {
-		return fmt.Errorf("components is not initialized")
-	}
-	components.Port = port
-
-	ccq.h, err = libp2p.New(
-		// Use the keypair we generated
-		libp2p.Identity(priv),
-
-		// Multiple listen addresses
-		libp2p.ListenAddrStrings(
-			components.ListeningAddresses()...,
-		),
-
-		// Enable TLS security as the only security protocol.
-		libp2p.Security(libp2ptls.ID, libp2ptls.New),
-
-		// Enable QUIC transport as the only transport.
-		libp2p.Transport(libp2pquic.NewTransport),
-
-		// Let's prevent our peer from having too many
-		// connections by attaching a connection manager.
-		libp2p.ConnectionManager(components.ConnMgr), // TODO: Can we use the same connection manager?
-
-		// Let this host use the DHT to find other hosts
-		libp2p.Routing(func(h host.Host) (routing.PeerRouting, error) {
-			ccq.logger.Info("Connecting to bootstrap peers", zap.String("bootstrap_peers", bootstrapPeers))
-
-			bootstrappers, _ := bootstrapAddrs(ccq.logger, bootstrapPeers, h.ID())
-
-			// TODO(leo): Persistent data store (i.e. address book)
-			idht, err := dht.New(ctx, h, dht.Mode(dht.ModeServer),
-				// This intentionally makes us incompatible with the global IPFS DHT
-				dht.ProtocolPrefix(protocol.ID("/"+networkID)),
-				dht.BootstrapPeers(bootstrappers...),
-			)
-			return idht, err
-		}),
-	)
-
-	if err != nil {
-		return fmt.Errorf("failed to create p2p: %w", err)
+	if h == nil {
+		return fmt.Errorf("h is not initialized")
```

Comment:
Can this be a more descriptive error? Preferably one that doesn't require reading the source and finding the definition of `h` in the function arg spec?